pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1293,6 +1293,9 @@ class Timestamp(_Timestamp):
         Unit used for conversion if ts_input is of type int or float. The
         valid values are 'D', 'h', 'm', 's', 'ms', 'us', and 'ns'. For
         example, 's' means seconds and 'ms' means milliseconds.
+
+        The result will be stored in nanoseconds, and the unit attribute
+        will be set as ``'ns'``.
```

Comment:
this is only the case for floats.  for ints we do keep the unit where possible

Reply:
I noticed this as well (haven't checked to see if its expected):

```
import pandas as pd

unit = "s"

ts1 = pd.to_datetime(1, unit=unit)
ts2 = pd.Timestamp(1, unit=unit)

print(f"{ts1.unit=}, {ts2.unit=}")

td1 = pd.to_timedelta(1, unit=unit)
td2 = pd.Timedelta(1, unit=unit)

print(f"{td1.unit=}, {td2.unit=}")
```

output: 
```
ts1.unit='ns', ts2.unit='s'
td1.unit='ns', td2.unit='ns'
```

Reply:
Thanks, clarified that it's only for floats